ecdsa/src/sp1.rs

```diff
@@ -75,18 +83,22 @@ where
     ///
     /// This function is a modified version of [`crate::hazmat::verify_prehashed`] with
     /// changes implemented to support SP1 acceleration.
-    pub fn verify_signature_secp256k1(
+
+    pub fn verify_signature_secp256(
         pubkey: &[u8; 65],
         msg_hash: &[u8; 32],
         signature: &Signature<C>,
         s_inverse: &Scalar<C>,
+        curve: Secp256Curve,
     ) -> bool {
         let mut pubkey_x_le_bytes = pubkey[1..33].to_vec();
         pubkey_x_le_bytes.reverse();
         let mut pubkey_y_le_bytes = pubkey[33..].to_vec();
         pubkey_y_le_bytes.reverse();
-        let affine =
-            Secp256k1Point::from_le_bytes(&[pubkey_x_le_bytes, pubkey_y_le_bytes].concat());
+        let affine = match curve {
+            Secp256Curve::K1 => Secp256k1Point::from_le_bytes(&[pubkey_x_le_bytes, pubkey_y_le_bytes].concat()),
+            Secp256Curve::R1 => Secp256r1Point::from_le_bytes(&[pubkey_x_le_bytes, pubkey_y_le_bytes].concat()),
+        };
```

Comment:
Given that `affine` is only used in the call to `multi_scalar_multiplication`, I'd move this logic to be inside of the `match` statement at the bottom so there's not two match statements.